### ISSUE TITLE
[Bug]  Doesn't work for Gmail

### ISSUE THREAD

**Issue by ANSI-C** [mrdokenny/Cookie-AutoDelete#1423]
### Acknowledgements

- [X] I acknowledge that I have read the above items

### Describe the bug

Gmail seems to escape the autodeletion, dunno if cause CAD is failing to delete something of cause if Gmail using same techniques of Discord (probably something more advanced than keeping session tokens also in memory and rewrite them to localStorage on beforeUnload event).

### To Reproduce

- none of Google related/owned domains are in grey or whitelist
- second(s) Delay Before Automatic Cleaning => 60 seconds
- ENABLED ALL following options, the others are disabled:
* Enable Automatic Cleaning
* Enable Cleanup on Domain Change
* Enable Greylist Cleanup on Browser Restart
* Clean All Expired Cookies
* Clean Existing Data for Newly Enabled Browsing Data Types
* Enable Cache Cleanup (Firefox 78+, Chrome 74+)
* Enable IndexedDB Cleanup (Firefox 77+, Chrome 74+)
* Enable LocalStorage Cleanup (Firefox 58+, Chrome 74+)
* Enable Plugin Data Cleanup (Firefox 78+, Chrome 74+)
* Enable Service Workers Cleanup (Firefox 77+, Chrome 74+)

1) keep opened just ONE tab/window
2) go to Gmail and authenticate
3) close the tab
4) wait 2 minutes
5) close the browser completely
6) reopen the browser
7) wait 2 minutes
8) go to Gmail
9) you will see that Gmail is still authenticated

### Expected Behavior

Since all kind of storage (cookies, indexedDB, cache, localStorage, etc etc) are supposed to be deleted by CAD, and given that we close the tab, releasing the page's variables in memory, and given that we even close the browser, waiting every time the double of the time set for auto-deletion, it is supposed that Gmail shouldn't have a way to keep any persistent info about logged in user. So it should present the user with a blank login form on reopen, instead it reloads the authenticated profile of the user. 

### System Info - Operating System (OS)

Linux 5.4.0-121-generic

### System Info - Browser Info

Vivaldi 5.3.2679.68 (Stable channel) stable (64-bit)

### System Info - CookieAutoDelete Version

3.8.1

### Additional Context

Vivaldi Revision ecfb656707bbec7aa16f80eb0ea2f4e98e14567b
JavaScript V8 10.2.154.13

**Comment by kennethtran93** [comment]
With gmail, the credentials may be stored in another domain (e.g. accounts.google.com) or something.  Also with Chrome, there are additional site data types we have yet to implement cleanup for as well, so data may be stored in one of those.

If you know where to navigate to where the site data is being stored, perhaps you can let me know what domain/site data item remains after cleanup. `chrome://settings/siteData` should show you the sites and data (+data type) being stored.

For example, looking at twitter, here's what is being stored on my chrome browser:
![image](https://user-images.githubusercontent.com/6724477/178194813-5de2708f-86f5-48b8-88e1-e9dff38bc7e0.png)
Cache Storage in this case is one of those site data types that we have yet to implement (already a feature request), so not a bug at this time.

In addition, with Chrome there's an option to allow an extension to run on all sites - this needs to be checked/enabled
--> https://support.google.com/chrome_webstore/answer/2664769

**Comment by ANSI-C** [comment]
Yes, I think the reason is that cookies are stored under accounts.google.com, so after login redirect CAD has no way to know it has to delete account.google.com when I close gmail.com tab. Actually from site content it is clear that no data is stored from gmail, but local storage and cookies data is stored from accounts.google.com.
PS: I noted that you set a cookie for every domain, why?

**Comment by kennethtran93** [comment]
> Yes, I think the reason is that cookies are stored under accounts.google.com, so after login redirect CAD has no way to know it has to delete account.google.com when I close gmail.com tab. Actually from site content it is clear that no data is stored from gmail, but local storage and cookies data is stored from accounts.google.com. PS: I noted that you set a cookie for every domain, why?

Currently, until I add a new permission that allows me to read all web requests made, we currently only clean those that have at least one cookie set for the domain, or by actually visiting the site.

As for why we set a cookie for every domain, it is the only way we can look through a list of domains that may contain something.  All other browsing site data requires webextensions to already know the domain/hostname to clean up beforehand.  This way, even if the website you're visiting doesn't set any cookie, we will still set one for that domain and can reference that to clean up during the cleanup cycle.

And due to the note I mentioned in the first paragraph, if a third-party domain doesn't set any cookies, we won't be able to detect and clean those yet.
